src/App/StarknetSyncState.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, PropsWithChildren, ReactElement, useEffect } from "react";
+import { Abi } from "starknet";
+import { useConnectors, useContract, useStarknet } from "@starknet-react/core";
+
+import config from "src/config";
+import profileRegistryAbi from "src/abis/profileRegistry.json";
+import { useSetRecoilState } from "recoil";
+import { accountAddressAtom, accountAtom, profileRegistryContractAtom } from "src/state";
+
+const StarknetSyncState: FC<PropsWithChildren> = ({ children }) => {
+  const setAccountAddress = useSetRecoilState(accountAddressAtom);
+  const setAccount = useSetRecoilState(accountAtom);
+  const setProfileRegistryContract = useSetRecoilState(profileRegistryContractAtom);
+
+  const { account: accountAddress } = useStarknet();
```

Comment:
Don't we get the account here already? What's the difference with the one here https://whimsical.com/retrospective-08-07-22-Mbz6Smf3CoC941eJW5zfkc?

Reply:
Not sure to understand your question.

What we could do is :
- we can choose to not set the `accountAddress` in a recoil atom  here but only set the `account` object
- the `accountAddress` would be used only to deduce the `account` object
- we create a selector to retrieve the `account` address

Reply:
Yes let's do this :)